### DISCUSSION TITLE
Extend composer repositories

### DISCUSSION THREAD

**Discussion by BjBue** [pimcore/pimcore#15000] [Q&A / Support] [answered]
Hello,

I try to install the datadirector from blackbit.
I get a zip-file and in the readme is the following description:
Alternatively if you received the plugin code as zip file, please upload the zip file to your server - e.g. create a folder `bundles` in the Pimcore root folder) and add the following to your `composer.json`:
`"repositories": [
    {
        "type": "artifact",
        "url": "./bundles/"
    }
]`

My composer looks like this and is working absolutly correct:
`"repositories": {
    "pimcore_enterprise": {
      "type": "composer",
      "url": "https://enterprise.repo.pimcore.com/"
    }
  }`

All changes gives me a non valid json error. 
My last try is this:
`"repositories": [
    {
      "pimcore_enterprise": {
        "type": "composer",
        "url": "https://enterprise.repo.pimcore.com/"
      },
      "type": "artifact",
      "url": "https://enterprise.repo.pimcore.com/"
    },
    {
      "type": "artifact",
      "url": "./bundles/"
    }
  ]`

But all fail with a composer error. Can anyone give me a hint, i have no more idea to solve this.
Thank you very much

**Comment by jdreesen** [comment, marked as answer]
You mixed the named and non-named syntax.

Try this:

```json
"repositories": { 
	"pimcore_enterprise": { "type": "composer", "url": "https://enterprise.repo.pimcore.com/" }, 
    "zipped_bundles": { "type": "artifact", "url": "./bundles/" } 
}
```

or without names (mind the different type of parenthesis):

```json
"repositories": [
	{ "type": "composer", "url": "https://enterprise.repo.pimcore.com/" }, 
    { "type": "artifact", "url": "./bundles/" } 
]
```
